product_default_image/models/product_template.py

```
@@ -0,0 +1,317 @@
+        """ Change product template image to specified to_type's image
+            or to_img_bg.
+
+            Args:
+                to_type (str): Value specified allowed from image_type field
+                    which the image should be targeted to. Accepted values are:
+
+                    GLOBAL
+                        * product_image field defined in res.company
+
+                    CATEGORY
+                        * product's categ_id field's image
+
+                    NONE
+                        * Deletes the image
+
+                    CUSTOM
+                        * Set to the image you specify using to_img_bg.
+                        Note that targeting to custom will prevent the
+                        image from automatically being changed in the
+                        future until deleted or manually re-mapped to a
+                        default image.
+
+                to_img (base64): Can be used as an override if the template
+                    images should be changed to a specific image.
+                    Image is resized to image big value and scaled
+                    down to medium and small values.
+
+                    Note that to_type is required even if to_img_bg
+                    arg is supplied.
+
+                in_cache (bool): Set True if calling this method to change the
+                    images in cache (helpful when calling from an onchange
+                    method in product.template)
+
+            Returns:
+                ProductTemplate: Recordset if tmpl images have been changed.
+                None: if any of the conditions below are True.
+
+        """
+        if any([
+                to_type not in (NONE, GLOBAL, CATEGORY, CUSTOM),
+                to_type == GLOBAL_CATEGORY,
+                to_type == CUSTOM and not to_img_bg,
+                to_type == NONE and to_img_bg,
+                not isinstance(in_cache, bool)]):
+            return
```

In at least some of these cases, it seems like you would want to raise an exception with a helpful error message, rather than just doing nothing and silently returning.

---

Yeah I was debating about that for awhile. I think perhaps it may be best to separate these into separate checks and raise errors accordingly.